google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/MessageMatcher.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub.v1;
+
+import com.google.pubsub.v1.PubsubMessage;
+import org.mockito.ArgumentMatcher;
+
+public class MessageMatcher implements ArgumentMatcher<PubsubMessage> {
```

Comment:
Call this MessageDataMatcher, as we are not matching on the whole message, just the data.

And in the constructor just take in the data, not the whole PubsubMessage.

Reply:
the test case - 
```
verify(mockMessageReceiverWithAckResponse, times(1))
        .receiveMessage(
            argThat(new MessageDataMatcher(TEST_MESSAGE2.getMessage().getData())),
            any(AckReplyConsumerWithResponse.class));
```
wont work if i convert this constructor to just take a data (type bytestring) since receiveMessage() takes in a PubsubMessage which is why i kept it as PubsubMessage

Reply:
It will work, you just need to change the MessageDataMatcher implementation to match on the byte content only:

```
public class MessageDataMatcher implements ArgumentMatcher<PubsubMessage> {

  private ByteString expectedData;

  public MessageDataMatcher(ByteString expectedData) {
    this.expectedData = expectedData;
  }

  @Override
  public boolean matches(PubsubMessage message) {
    return expectedData.equals(message.getData());
  }
}
```